stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java

```diff
@@ -0,0 +1,152 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.mime.nativeimpl;
+
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import io.netty.handler.codec.http.HttpHeaders;
+import org.ballerinalang.jvm.BallerinaErrors;
+import org.ballerinalang.jvm.types.BArrayType;
+import org.ballerinalang.jvm.types.BTypes;
+import org.ballerinalang.jvm.values.ArrayValue;
+import org.ballerinalang.jvm.values.ArrayValueImpl;
+import org.ballerinalang.jvm.values.HandleValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.mime.util.MimeUtil;
+
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import static org.ballerinalang.mime.util.MimeConstants.ENTITY_HEADERS;
+import static org.ballerinalang.mime.util.MimeConstants.HEADER_NOT_FOUND;
+
+/**
+ * Utilities related to entity headers.
+ *
+ * @since 1.1.0
+ */
+public class EntityHeaders {
+    public static void addHeader(ObjectValue entityObj, String headerName, String headerValue) {
+        if (headerName == null || headerValue == null) {
+            return;
+        }
+        HttpHeaders httpHeaders;
+        httpHeaders = getHeaders(entityObj);
+        httpHeaders.add(headerName, headerValue);
+    }
+
+    public static String getHeader(ObjectValue entityObj, String headerName) {
+        if (entityObj.getNativeData(ENTITY_HEADERS) == null) {
+            throw BallerinaErrors.createError(HEADER_NOT_FOUND, "Http header does not exist");
+        }
+        HttpHeaders httpHeaders = (HttpHeaders) entityObj.getNativeData(ENTITY_HEADERS);
+        if (httpHeaders != null && httpHeaders.get(headerName) != null && !httpHeaders.get(headerName).isEmpty()) {
+            return httpHeaders.get(headerName);
+        } else {
+            throw BallerinaErrors.createError(HEADER_NOT_FOUND, "Http header does not exist");
+        }
+    }
+
+    public static ArrayValue getHeaderNames(ObjectValue entityObj) {
+        ArrayValue stringArray = new ArrayValueImpl(new BArrayType(BTypes.typeHandle));
```

Comment:
IINM, There is a new constructor for creating these values. BValueCreator or similar. We should use those. @vinok88

Reply:
Yes, We have deprecated the ArrayValueImpl class. Please use the BValueCreator.createXXX methods.